### PR TITLE
Improve coverage and minor bug fixes

### PR DESCRIPTION
**Pull request by abhisrkckl**
Here are some more tests to improve coverage.

I have also fixed some bugs in `zima` and `toa`.

### CODE REVIEW

File: src/pint/orbital/kepler.py [hunk 1]
```diff
@@ -146,8 +146,8 @@ def kepler_2d(params, t):
     eps1 = params.eps1
     eps2 = params.eps2
     t = t - params.t0
-    if eps1 == 0 and eps2 == 0:
-        eps1 = 1e-50
+    # if eps1 == 0 and eps2 == 0:
```
Comment (abhisrkckl) [inline src/pint/orbital/kepler.py:149]: This function works fine even without this.

File: src/pint/pulsar_ecliptic.py [hunk 2]
```diff
@@ -52,21 +52,6 @@ class PulsarEcliptic(coord.BaseCoordinateFrame):
...
-                kwargs["d_lon_coslat"] = kwargs["pm_lon_coslat"]
-                del kwargs["pm_lon_coslat"]
-            except KeyError:
-                pass
-            try:
-                kwargs["d_lat"] = kwargs["pm_lat"]
-                del kwargs["pm_lat"]
-            except KeyError:
-                pass
-
```
Comment (abhisrkckl) [inline src/pint/pulsar_ecliptic.py:69]: This is no longer needed as PINT now requires astropy>=4

File: src/pint/toa.py [hunk 3]
```diff
@@ -771,7 +771,7 @@ def read_toa_file(filename, process_includes=True, cdict=None, dir=None):
                 cdict[cmd] = float(d["Command"][1]) * u.MHz
             elif cmd in ("EFAC", "PHA1", "PHA2"):
                 cdict[cmd] = float(d["Command"][1])
-                if cmd in ("PHA1", "PHA2", "TIME", "PHASE"):
```
Comment (abhisrkckl) [inline src/pint/toa.py:774]: These values are not reachable.

**Comment by abhisrkckl** [comment]
Done with this PR for now.

File: src/pint/toa.py (hunk 3 above)
Comment (abhisrkckl) [reply src/pint/toa.py:774]: TIME and PHASE commands are treated a few lines above here. I think it is OK to remove this.

File: tests/test_reduced_precision.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import numpy as np
+from pint.utils import require_longdouble_precision, PINTPrecisionError
+import pytest
+
+
+def test_require_longdouble_precision():
+    ld = np.longdouble
+    np.longdouble = np.float128 = np.float64
```
[the remark this answers is not in the document]
Comment (abhisrkckl) [reply tests/test_reduced_precision.py]: Done.

**Comment by dlakaplan** [comment]
For the `zima` plotting tests, do you need anything like:
```
@pytest.mark.skipif(
    "DISPLAY" not in os.environ, reason="Needs an X server, xvfb counts"
)
```
I guess it passed CI, so maybe not.

**Comment by dlakaplan** [comment]
OK, good.  I can merge if you are ready.

**Comment by abhisrkckl** [comment]
Yes. I think this is ready to go.

**Pull request merged**